### DISCUSSION TITLE
How can I exempt a certain route/endpoint from gzip/brotli middleware?

### DISCUSSION THREAD

**Discussion by erichaus** [encode/starlette#1661] [Q&A] [answered]
I have added [Brotli-Asgi](https://github.com/fullonic/brotli-asgi) middleware to my Starlette api but for a specific route that returns a zip file via StreamingResponse, I'd like to exclude the brotli compression. Is this possible?

**Comment by Kludex** [comment, marked as answer]
Now, it's possible to apply middlewares by `Router`, `Route`, `Mount`, and others.

See https://www.starlette.io/middleware/#applying-middleware-to-groups-of-routes for more details.
